Exercise 1 BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,214 @@
+package com.tridiumuniversity.myFile;
+
+import javax.baja.nre.annotations.NiagaraAction;
+import javax.baja.nre.annotations.NiagaraProperty;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.*;
+
+@NiagaraType
+@NiagaraProperty(
+        name = "fileName",
+        type = "BString",
+        defaultValue = "BString.DEFAULT"
+)
+@NiagaraProperty(
+        name = "path",
+        type = "BString",
+        defaultValue = "BString.DEFAULT"
+)
+@NiagaraProperty(
+        name = "size",
+        type = "BInteger",
+        defaultValue = "BInteger.DEFAULT"

```

Comment:
Using `BString` or `BInteger` as the `type` and then specifying the `defaultValue` as `BString.DEFAULT` or `BInteger.DEFAULT` is perfectly fine. Just so that you're aware, you could have also used Java primitives:

```
@NiagaraProperty(
  name = "fileName",
  type = "String",
  defaultValue = "\"\""
)
@NiagaraProperty(
  name = "path",
  type = "String",
  defaultValue = "\"\""
)
@NiagaraProperty(
  name = "size",
  type = "int",
  defaultValue = "0"
)
```